app/helpers/blacklight_helper.rb

```diff
@@ -269,43 +271,19 @@ def truncated_link(doc, field, opts = { counter: nil }, length = 200)
 
     ##
     # Check if any search parameters have been set
-    # @return [Boolean]
-    def search_parameters?
-      blacklight_params[:q].present? || blacklight_params[:f].present? || blacklight_params[:search_field].present?
-    end
-
-    # Determines whether or not a query is provided in the request
-    # @param [ActionController::Parameters] params
-    # @return [Boolean]
-    def empty_search_query?(params)
-      params[:q].blank?
-    end
-
-    # Determines whether or not the request is scoped for all fields
-    # @param [ActionController::Parameters] params
-    # @return [Boolean]
-    def all_search_fields_query?(params)
-      params[:search_field] && params[:search_field] == 'all_fields'
-    end
-
-    # Determines whether or not the request is scoped for a specific page number
-    # @param [ActionController::Parameters] params
-    # @return [Boolean]
-    def first_page_query?(params)
-      !params[:page]
-    end
-
-    # Determines whether or not the request facets upon any fields
     # @param [ActionController::Parameters] params
     # @return [Boolean]
-    def faceted_query?(params)
-      params[:f]
+    def search_parameters?(params)
+      params[:q].present? || params[:f].present?
     end
 
-    # Determines whether or not this is a default query
+    # Determines whether or not the user is requesting an excessively high page of results
     # @param [ActionController::Parameters] params
     # @return [Boolean]
-    def default_search_query?(params)
-      empty_search_query?(params) && all_search_fields_query?(params) && first_page_query?(params) && !faceted_query?(params)
+    def excessive_paging?(params)
+      page = params[:page].to_i || 0
+      return false if page == 0
+      return false if search_paramters?(params) && page < 1000
```

Comment:
`search_paramters?(params)` is missing an `e`

Reply:
👍 I've fixed that typo and added a static 400 Bad Request error page so the display is better.